### PR TITLE
Add documentation for reverse-proxy use

### PR DESCRIPTION
**Pull request by nils-van-zuijlen**
### Component/Part
Documentation

### Description
This PR adds documentation for reverse-proxy setup.

### Steps

- [x] added implementation
- [x] added / updated tests
- [x] added / updated documentation
- [x] extended changelog

### Related Issue(s)
#256

### CODE REVIEW

**Review by ErikMichelson — COMMENTED** [review]
That looks pretty complete and well-structured! Thank you for this contribution! 👍

Just one question: Did you test this setup to be running as expected (especially the nginx config)?

**Comment by nils-van-zuijlen** [comment]
Yes, that's the setup I run myself (with a different port, but 3000 is HedgeDoc's default, so I used that there).

File: docs/setup/reverse-proxy.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+## Reverse Proxy config
+
+### Generic
+
+The reverse proxy must allow websocket `Upgrade` requests at path `/sockets.io/`.
+
+It must pass through the scheme used by the client (http or https).
+
+### nGinx
```
Comment (ErikMichelson) [inline docs/setup/reverse-proxy.md:32]: Just seen this little thing: I know the software so far only as "nginx" or "NGINX". Any reason for the capital letter G? 😅

Comment (nils-van-zuijlen) [reply docs/setup/reverse-proxy.md:32]: I think I was thinking about the last style mentioned on Wikipedia,
image
but my mind played on me :sweat_smile:

**Review by ErikMichelson — APPROVED** [review]

**Pull request merged**